### PR TITLE
Add lift_flux

### PR DESCRIPTION
**Pull request by wthrowe**

### CODE REVIEW

File: src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/// \file
+/// Defines function lift_flux.
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Variables.hpp"
+
+/// \ingroup DiscontinuousGalerkin
```
Comment (kidder) [inline src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp]: put in namespace dg

File: src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/// \brief Lifts fluxes from an interface to a volume contribution.
+///
+/// This does a diagonal lift.  This is apparently an approximation.
```
Comment (kidder) [inline src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp]: This does a diagonal lift, as we are choosing the integration quadrature points to be the same as the collocation points.  We do this as it is much faster, but it does decrease the accuracy of the method.  See Saul's paper.

File: tests/Unit/Numerical/DiscontinuousGalerkin/Test_LiftFlux.cpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+struct Var {
+  using type = Scalar<DataVector>;
+};
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.DiscontinuousGalerkin.LiftFlux", "[Unit][Numerical]") {
+  const size_t perpendicular_extent = 5.;
```
Comment (kidder) [inline tests/Unit/Numerical/DiscontinuousGalerkin/Test_LiftFlux.cpp]: huh?

File: tests/Unit/Numerical/DiscontinuousGalerkin/Test_LiftFlux.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  const auto coordinate_map = make_coordinate_map<Frame::Logical, Frame::Grid>(
+      CoordinateMaps::ProductOf2Maps<CoordinateMaps::AffineMap,
+                                     CoordinateMaps::AffineMap>(
+      {-1., 1., -5., 7.}, {-1., 1., 2., 5.}));
+  const double element_length = 3.;
```
Comment (kidder) [inline tests/Unit/Numerical/DiscontinuousGalerkin/Test_LiftFlux.cpp]: compute this

**Review by kidder — CHANGES_REQUESTED** [review]
squash

File: src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp (hunk 2 above)
Comment (fmahebert) [reply src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp]: Here's a possible Doxy comment. I tried to say just enough to point the curious reader towards more detail, without actually getting involved in any technicalities. The ideal comment may have more (or less) detail than what I've written...

```
\brief Lifts the flux contribution from an interface to the volume.

The lifting operation takes the (d-1)-dimensional flux term at the
interface and computes the corresponding d-dimensional term in the
volume. SpECTRE implements an efficient DG method in which each
interface grid point contributes only to that same grid point of the
volume. 

\details
SpECTRE implements a DG method with a diagonalized mass matrix (also
known as a mass-lumping scheme). This choice gives a large
reduction in the computational cost of the lifting operation, however,
the scheme is slightly less accurate, especially when the grid is
deformed by non-trivial Jacobians. For more details on the
diagonalization of the mass matrix and its implications, see Saul's
DG formulation paper, especially Section 3:
https://arxiv.org/abs/1510.01190.

\note
(keep Will's note)
```

**Review by kidder — DISMISSED** [review]

**Comment by wthrowe** [comment]
I found a minor bug in this.  Please hold off on merging.

**Comment by wthrowe** [comment]
Changes from reviewed version:
```diff
--- a/src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp
+++ b/src/Numerical/DiscontinuousGalerkin/LiftFlux.hpp
@@ -40,7 +40,12 @@ Variables<Tags> lift_flux(const Variables<Tags>& local_flux,
   lift_factor *= -0.5 * (extent_perpendicular_to_boundary *
                          (extent_perpendicular_to_boundary - 1));
 
-  auto& lifted_data = numerical_flux;
+  // Using a reference (as for lift_factor) wouldn't gain us anything
+  // here because copy elision doesn't apply to function parameters.
+  // Returning from a reference value also triggers a copy, so we
+  // would have to explicitly move in the return statement or return
+  // numerical_flux to avoid that.
+  auto lifted_data = std::move(numerical_flux);
   lifted_data -= local_flux;
   lifted_data *= lift_factor;
   return lifted_data;
```

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**